### ISSUE TITLE
About the four direction diffusion

### ISSUE THREAD

**Issue by chenchr** [Liusifei/pytorch_spn#1]
Hello. Thanks for sharing the aewsome works. After reading the paper and the pytorch code, I have a question about the four direction diffusion. As the caffe prototxt is too complex, I guess that you doing the four direction diffusion parallelly, following a elementwise max operation to produce the final refined result? I am not sure whether I am right. Could you please explain more about that ?  The core question is that ``` is four direction diffusion refine in parallel or sequentially‘’‘ ’？ Thank you very much!

**Comment by Liusifei** [comment]
Sorry that i did not check it earlier. Yes it is in parallel and integrated then by node-wise maxpooling.

**Comment by chenchr** [comment]
Thank you for the reply.

**Comment by ZhangC2** [comment]
Hello. After reading the paper, I have a question about experiment. For HELEN dataset, training a SPN with patches of input rgb image and coarse segmentation mask. How about Pascal VOC dataset? it is trained with patches as HELEN dataset did?  Thank you very much! @Liusifei @
